src/ls/definition.cr

```diff
@@ -15,84 +15,30 @@ module Mint
           stack =
             server.nodes_at_cursor(params)
 
-          html_style(server, workspace, stack) ||
-            html_attribute(server, workspace, stack) ||
-            html_component(server, workspace, stack)
+          if node = stack[0]?
+            definition(node, server, workspace, stack)
+          end
         end
       end
 
-      def with_stack(stack : Array(Ast::Node), &)
-        yield StackReader.new(stack)
+      def definition(node : Ast::Node, server : Server, workspace : Workspace, stack : Array(Ast::Node))
+        nil
       end
 
-      def selection(location : Ast::Node::Location) : LSP::Range
-        LSP::Range.new(
-          start: LSP::Position.new(
-            line: location.start[0] - 1,
-            character: location.start[1]
-          ),
-          end: LSP::Position.new(
-            line: location.end[0] - 1,
-            character: location.end[1]
-          )
-        )
-      end
-
-      # Returns the range for the name part for a node
-      def selection(node : Ast::Node) : LSP::Range
-        selection(node.location)
-      end
-
-      def selection(node : Ast::Component) : LSP::Range
-        # Select only the name part of the component
-        #   global component MintComponent {
-        #                    ^^^^^^^^^^^^^
-        selection(node.name)
-      end
-
-      def selection(node : Ast::HtmlAttribute) : LSP::Range
-        # Select only the name part of the attribute
-        # <Component attribute={value}>
-        #            ^^^^^^^^^
-        selection(node.name)
-      end
-
-      def selection(node : Ast::HtmlComponent) : LSP::Range
-        # Select only the name part of the component
-        # <Component attribute={value}>
-        #  ^^^^^^^^^
-        selection(node.component)
+      def cursor_intersects?(node : Ast::Node, position : LSP::Position) : Bool
+        node.location.contains?(position.line + 1, position.character)
```

Comment:
The tests are passing, so it must be right just curious why `+ 1`?

Reply:
The real answer is that I just copied it from [`nodes_at_cursor`](https://github.com/mint-lang/mint/blob/master/src/ls/server.cr#L37) ... :sweat_smile:  

But also, it's because for `node.location`'s the line number [is 1-based ](https://github.com/mint-lang/mint/blob/master/src/ast/node.cr#L103), but in the language server specification both line and character are [zero based](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#position)

Reply:
Yeah, I had a feeling 😅 thanks for confirming 🙏